cpp/src/reductions/reduction.cuh

```diff
@@ -40,9 +40,9 @@ namespace detail {
  * @param[in] num_items the number of items
  * @param[in] op        the reduction operator
  * @param[in] stream    cuda stream
- * @returns   scalar    output scalar in device memory
+ * @returns   unique_ptr<scalar>    output scalar in device memory
```

Comment:
I don't think you need to put the type of the return here, just the description. The type should be obvious from the function signature, which is included in any rendered documentation. http://www.doxygen.nl/manual/commands.html#cmdreturns